libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -277,4 +283,43 @@ public void onServiceDisconnected(ComponentName className) {
       serviceBound = false;
     }
   };
+
+  private boolean checkLocationPermission() {
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
+      return true;
+    } else {
+      permissionBackoff();
+      return false;
+    }
+  }
+
+  private void permissionBackoff() {
+    Handler handler = new Handler();
+    ExponentialBackoff counter = new ExponentialBackoff();
+
+    PermissionCheckRunnable permissionCheckRunnable = new PermissionCheckRunnable(context);
+
+    long nextWaitTime = counter.nextBackOffMillis();
+    handler.postDelayed(permissionCheckRunnable, nextWaitTime);
+  }
+
+  private final class PermissionCheckRunnable implements Runnable {
```

Comment:
Why is `PermissionCheckRunnable` a nested class of `MapboxTelemetry`? Is it necessary? Could we extract `PermissionCheckRunnable` class into a `PermissionCheckRunnable.java` file/class?